cli/internal/env/env_test.go

```diff
@@ -1,297 +1,225 @@
+			want: EnvironmentVariableMap{
+				"ALL":      "VALUE",
+				"OF":       "VALUE",
+				"THESE":    "VALUE",
+				"ARE":      "VALUE",
+				"INCLUDED": "VALUE",
 			},
-			want: EnvironmentVariablePairs{"TURBOREPO=true"},
+			wantErr: false,
 		},
 		{
-			env:  []string{"NEXT_PUBLIC_MY_VERCEL_URL=me.vercel.com", "TURBOREPO=true", "TURBO_CI_VENDOR_ENV_KEY=NEXT_PUBLIC_VERCEL_"},
-			name: "$TURBO_CI_VENDOR_ENV_KEY excludes automatically added env vars",
-			args: args{
-				envKeys:     []string{"TURBOREPO"},
-				envPrefixes: []string{"NEXT_PUBLIC"},
+			name: "include everything, exclude everything",
+			self: EnvironmentVariableMap{
+				"ALL":      "VALUE",
+				"OF":       "VALUE",
+				"THESE":    "VALUE",
+				"ARE":      "VALUE",
+				"EXCLUDED": "VALUE",
 			},
-			want: EnvironmentVariablePairs{"NEXT_PUBLIC_MY_VERCEL_URL=me.vercel.com", "TURBOREPO=true"},
+			wildcardPatterns: []string{"*", "!*"},
+			want:             EnvironmentVariableMap{},
+			wantErr:          false,
 		},
 		{
-			env:  []string{"TURBO_CI_VENDOR_ENV_KEY_VAL=true", "TURBO_CI_VENDOR_ENV_KEY=TURBO_CI_VENDOR_ENV_KEY"},
-			name: "$TURBO_CI_VENDOR_ENV_KEY should not exclude itself",
-			args: args{
-				envKeys:     []string{},
-				envPrefixes: []string{"TURBO_"},
+			name: "include everything, exclude one",
+			self: EnvironmentVariableMap{
+				"ONE":      "VALUE",
+				"OF":       "VALUE",
+				"THESE":    "VALUE",
+				"IS":       "VALUE",
+				"EXCLUDED": "VALUE",
 			},
-			want: EnvironmentVariablePairs{},
+			wildcardPatterns: []string{"*", "!EXCLUDED"},
+			want: EnvironmentVariableMap{
+				"ONE":   "VALUE",
+				"OF":    "VALUE",
+				"THESE": "VALUE",
+				"IS":    "VALUE",
+			},
+			wantErr: false,
 		},
 		{
-			env:  []string{"NEXT_PUBLIC_VERCEL_ENV=true", "MANUAL=true", "TURBO_CI_VENDOR_ENV_KEY=_VERCEL_"},
-			name: "blocked env var is allowed if manually specified",
-			args: args{
-				envKeys:     []string{"NEXT_PUBLIC_VERCEL_ENV", "MANUAL"},
-				envPrefixes: []string{"NEXT_PUBLIC_"},
+			name: "include everything, exclude a prefix",
+			self: EnvironmentVariableMap{
+				"EXCLUDED_SHA":  "VALUE",
+				"EXCLUDED_URL":  "VALUE",
+				"EXCLUDED_USER": "VALUE",
+				"EXCLUDED_PASS": "VALUE",
+				"THIS":          "VALUE",
+				"IS":            "VALUE",
+				"INCLUDED":      "VALUE",
+			},
+			wildcardPatterns: []string{"*", "!EXCLUDED_*"},
+			want: EnvironmentVariableMap{
+				"THIS":     "VALUE",
+				"IS":       "VALUE",
+				"INCLUDED": "VALUE",
 			},
-			want: EnvironmentVariablePairs{"MANUAL=true", "NEXT_PUBLIC_VERCEL_ENV=true"},
+			wantErr: false,
 		},
 	}
-
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			// set the env vars
-			setEnvs(tt.env)
-			// test
-			res, err := GetHashableEnvVars(tt.args.envKeys, tt.args.envPrefixes, "TURBO_CI_VENDOR_ENV_KEY")
-			if err != nil {
-				t.Errorf("error setup failure: %s", err)
+			got, err := tt.self.FromWildcards(tt.wildcardPatterns)
+			if (err != nil) != tt.wantErr {
```

Comment:
I tried to create an error, but since I'm using `QuoteMeta` and only adding `.*` the only point of error (`regexp.Compile`) basically can't error unless I authored something wrong. Will remove.